############
## Debug settings
############
debug_nbv_main_states: true
debug_mapping: false
debug_view_generator: false
debug_view_selecter: false
debug_save_state: false
debug_load_state: true
############
## Scenario settings
############
nav_bounds_x_max: 20
nav_bounds_x_min: -20
nav_bounds_y_max: 20
nav_bounds_y_min: -20
nav_bounds_z_max: 20
nav_bounds_z_min: 0.7 ## Do not allow the UAV to go below this height, as it may collide with the floor

object_bounds_x_max: 20
object_bounds_x_min: -20
object_bounds_y_max: 20
object_bounds_y_min: -20
object_bounds_z_max: 10
object_bounds_z_min: 0


############
## Vehicle settings
############
# 0: floating sensor
# 1: iris
vehicle_type: 0

## Number of RBGD cameras used
camera_count: 2

## UAV dimensions (used in collision detection)
uav_obstacle_distance_min: 4
uav_collision_radius: 0.8

## UAV Poses
uav_pose_x_0: 5
uav_pose_y_0: -5
uav_pose_z_0: .5
uav_pose_yaw_0: 1.571

uav_pose_x_1: 0
uav_pose_y_1: -20
uav_pose_z_1: 5
uav_pose_yaw_1: 1.571

uav_pose_x_2: -3
uav_pose_y_2: 0
uav_pose_z_2: 5
uav_pose_yaw_2: 0

uav_pose_x_3: 10
uav_pose_y_3: 10
uav_pose_z_3: 5
uav_pose_yaw_3: 0.3

############
## Mapping settings
############
mapping_integrate_prediction: false #if true, put prediction in final map. If false, put prediction in separate map
mapping_integrate_occupancy: 0.7 # Occupancy value for predicted cells
mapping_octree_resolution: 0.2
mapping_occupancy_threshold: 0.6
mapping_save_map_iterations: 20 #save map every "x"iterations in case of crash
mapping_sensor_data_min_height_: 0.3 #Min height from floor in order to register points
mapping_voxel_grid_res_profile: 0.1
mapping_voxel_grid_res_rgbd: 0.02

############
## Profiling settings
############
profiling_skip: true
profiling_skip_load_map: false #if profiling was skipped, load a preexisting map?
profiling_check_symmetry: true
profiling_fill_octomap: true  #Create an octomap in the initial scans?
profiling_fill_octomap_continuously: false
profiling_use_symmetry: true
profiling_scan_speed: 1.0

# 0: adaptive circular
# 1: bounded box
profiling_method: 0
profiling_circular_waypoints: 5

############
## Termination settings
############
# 0: Max iterations
# 1: Global Entropy Change Percentage Difference
# 2: Local Entropy Change Per Voxel
# 3: Utility Threshold
termination_type: 0
termination_ros_spin_afterwards: false
termination_window_size: 5
termination_repeat_window_size: 10 #Number of permissible repetitions before the UAV is considered stuck in a local minima

termination_iterations_max: 500
termination_entropy_global_min_change_threshold: 0.0001
termination_entropy_local_min_change_threshold: 0.001
termination_utility_min_threshold: 0.05

############
## View Generation settings
############
# 0: Nearest Neighbor (NN)
# 1: Adaptive NN
# 2: Frontier
# 3: Adaptive NN followed by Frontier at local minima
view_generator_type: 2
view_generator_nn_pos_res_x: 1.0
view_generator_nn_pos_res_y: 1.0
view_generator_nn_pos_res_z: 1.0
view_generator_nn_pos_res_yaw: rad(pi/4)
view_generator_nn_adaptive_local_minima_threshold: 0.005
view_generator_nn_adaptive_local_minima_iterations: 3
view_generator_nn_adaptive_utility_threshold: 0.1
view_generator_frontier_density_threshold: 100
view_generator_frontier_local_minima_threshold: 0.002
view_generator_frontier_local_minima_iterations: 4
view_generator_frontier_minimum_size: 5
view_generator_frontier_nearest_count: 50
view_generator_frontier_cylinder_radius: 3.5
view_generator_frontier_cylinder_height: 1


############
## View Selecter settings
############
# 0: Information Gain (IG)
# 1: Information Gain with exponential Distance
# 2: Point Density
# 10: Proposed
# 11: Proposed (ray length)
view_selecter_type: 10
view_selecter_compare: false
view_selecter_compare_type: 0

view_selecter_must_see_occupied: true #if true, views without a single occupied cell are given no utility
view_selecter_ignore_entropies_at_clamping_points: true
view_selecter_weight_distance: 0.5

view_selecter_proposed_weight_entropy:    0
view_selecter_proposed_weight_density:    0.5
view_selecter_proposed_weight_prediction: 0
view_selecter_proposed_weight_distance:   0.00
view_selecter_proposed_use_node_count: true
